subsys/bluetooth/host/hci_core.c

```diff
@@ -147,25 +147,38 @@ static struct cmd_data cmd_data[CONFIG_BT_HCI_CMD_COUNT];
 /* HCI command buffers. Derive the needed size from BT_BUF_RX_SIZE since
  * the same buffer is also used for the response.
  */
-#define CMD_BUF_SIZE BT_BUF_RX_SIZE
+#define MIN_CMD_BUF_SIZE BT_BUF_SIZE(                     \
+	sizeof(struct bt_hci_evt_cmd_complete) +          \
+	sizeof(struct bt_hci_rp_read_supported_commands))
+
+#define CMD_BUF_SIZE MIN(BT_BUF_RX_SIZE, MIN_CMD_BUF_SIZE)
 NET_BUF_POOL_FIXED_DEFINE(hci_cmd_pool, CONFIG_BT_HCI_CMD_COUNT,
 			  CMD_BUF_SIZE, NULL);
 
 NET_BUF_POOL_FIXED_DEFINE(hci_rx_pool, CONFIG_BT_RX_BUF_COUNT,
 			  BT_BUF_RX_SIZE, NULL);
 
 #if defined(CONFIG_BT_CONN)
+#if defined(BT_CTLR)
+#define NUM_COMPLETE_MAX_HANDLES 1
+#else
+#define NUM_COMPLETE_MAX_HANDLES 10
```

Comment:
Why not CONFIG_BT_MAX_CONN? I like that better than an arbitrary magic 10.

Reply:
🤦‍♂ Yes, that would make sense.